packages/flutter/lib/src/material/list_item.dart

```diff
@@ -5,53 +5,123 @@
 import 'package:flutter/widgets.dart';
 
 import 'ink_well.dart';
+import 'theme.dart';
 
+/// Material List items are one to three lines of text optionally flanked by icons.
+/// Icons are defined with the [left] and [right] parameters. The first line of text
+/// is not optional and is specified with [primary]. The value of [secondary] will
+/// occupy the space allocated for an aditional line of text, or two lines if
+/// isThreeLine: true is specified. If isDense: true is specified then the overall
+/// height of this list item and the size of the DefaultTextStyles that wrap
+/// the [primary] and [secondary] widget are reduced.
 class ListItem extends StatelessComponent {
   ListItem({
     Key key,
     this.left,
-    this.center,
+    this.primary,
+    this.secondary,
     this.right,
+    this.isThreeLine: false,
+    this.isDense: false,
     this.onTap,
     this.onLongPress
   }) : super(key: key) {
-    assert(center != null);
+    assert(primary != null);
+    assert(isThreeLine ? secondary != null : true);
   }
 
   final Widget left;
-  final Widget center;
+  final Widget primary;
+  final Widget secondary;
   final Widget right;
+  final bool isThreeLine;
+  final bool isDense;
   final GestureTapCallback onTap;
   final GestureLongPressCallback onLongPress;
 
+  TextStyle primaryTextStyle(BuildContext context) {
```

Comment:
I did this so that it would be easy to override the primary text style in a subclass.  Like:
```
MyListItem extends ListItem {
  TextStyle primaryTextStyle(BuildContext context) => super.primaryTextStyle(context).copyWith(...);
}
```